internal/provider/kubernetes/predicates.go

```diff
@@ -636,12 +636,26 @@ func (r *gatewayAPIReconciler) updateStatusForGatewaysUnderGatewayClass(ctx cont
 	}
 
 	for _, gateway := range gateways.Items {
-		r.updateStatusForGateway(ctx, &gateway)
+		r.updateGatewayStatus(&gateway)
 	}
 
 	return nil
 }
 
+// updateGatewayStatus triggers a status update for the Gateway.
+func (r *gatewayAPIReconciler) updateGatewayStatus(gateway *gwapiv1.Gateway) {
+	// The status added to GatewayStatuses is solely used to trigger the status updater
+	// and does not reflect the real changed status.
+	//
+	// The status updater will check the Envoy Proxy service to get the addresses of the Gateway,
+	// and check the Envoy Proxy Deployment/DaemonSet to get the status of the Gateway workload.
+	//
+	// Since the status does not reflect the actual changed status, we need to delete it first
+	// to prevent it from being considered unchanged. This ensures that subscribers receive the update event.
+	r.resources.GatewayStatuses.Delete(utils.NamespacedName(gateway))
```

Comment:
its unclear why the `Delete` is needed here
If the `Status` value is same as the previous value, it should be skipped, else updated

Reply:
The status added to GatewayStatuses here is solely used to trigger the status updaterand does not reflect the real changed status.

The status updater `updateStatusForGateway` will check the Envoy Proxy service to get the addresses of the Gateway, and check the Envoy Proxy Deployment/DaemonSet to get the status of the Gateway workload.

I plan to refactory this logic in a follow-up PR.